### PR TITLE
Update governance to establish steering committee and increase contributor diversity

### PR DESCRIPTION
**Pull request by jbw976**
### Description of your changes
This PR updates the Crossplane project governance to meet the following goals:

* Increase diversity of maintainers, enable more organizations to collaborate and have ownership
* Allow varying permissions and access across multiple repos and multiple maintainers
* Improve change velocity by expanding number of maintainers that can approve and merge pull requests
* Formalize responsibilities at the top level for the purpose of project stewardship, adoption, growth and health of the community, and strategic direction

This governance update includes the following high level changes:

* Institute a steering committee at the organization level responsible for overall project direction and health, as well as conflict resolution
* Define a bootstrapping and election process for filling the steering committee with members
* Enable each repository to define its own maintainer team and membership to increase the efficiency of accepting new code changes and instill a sense of ownership more broadly 
* Allow new repositories to be bootstrapped with maintainers that do not meet the minimum time length requirements
* Empower each maintainer team to perform conflict resolution and voting for issues within their scope

### Approval

As per the existing Crossplane governance, this update requires the existing maintainer team to have at least a 2/3 approval in order to ratify and merge.

### Checklist
I have:
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Ensured this PR contains a neat, self documenting set of commits.
- [x] Updated any relevant [documentation] and [examples].
- [x] Reported all new error conditions into the log or as an event, as
  appropriate.

For more about what we believe makes a pull request complete, see our
[definition of done].

[documentation]: https://github.com/crossplane/crossplane/tree/master/docs
[examples]: https://github.com/crossplane/crossplane/tree/master/cluster/examples
[definition of done]: https://github.com/crossplane/crossplane/tree/master/design/one-pager-definition-of-done.md

[skip ci]

### CODE REVIEW

File: OWNERS.md [hunk 1]
```diff
@@ -1,31 +1,33 @@
...
-## Senior maintainers
-
-Here's the current list of senior maintainers:
+## Steering Committee
 
 * Bassam Tabbara <bassam@upbound.io> ([bassam](https://github.com/bassam))
-  * Catch-all, architecture, strategy
-* Jared Watts <jared@upbound.io> ([jbw976](https://github.com/jbw976))
-  * Stacks architecture, release engineering
+  * 2 year term expires `2022-05-xy`
```
Comment (jbw976) [inline OWNERS.md:13]: note these exact dates should be filled in before merging, 2 years from date of acceptance and merge

File: GOVERNANCE.md [hunk 2]
```diff
@@ -1,133 +1,315 @@
...
+not be diluted too quickly while the project is still being bootstrapped and transitioned.
+
+Therefore, the membership will be bootstrapped over a **2 year period**, from the date of acceptance
+of this governance, as follows:
+
+* Membership will be **limited to 5** total seats
+* **3 seats** are granted to the senior maintainers from the existing Crossplane governance: Bassam
+    Tabbara, Jared Watts, and Nic Cope
+  * **2 year** term
+* **2 seats** to be filled by members of the community selected by the above committee members
```
Comment (jbw976) [inline GOVERNANCE.md:46]: we are in the process of confirming members for these two seats and will update this PR once they have been confirmed

File: OWNERS.md [hunk 3]
```diff
@@ -1,31 +1,33 @@
...
+  * 1 year term expires `2021-05-xy`
 
 ## Maintainers
 
-There are currently no maintainers. Maintainers will be added according to the [process for adding a maintainer](GOVERNANCE.md#becoming-a-maintainer) in the [governance](GOVERNANCE.md).
+* Bassam Tabbara <bassam@upbound.io> ([bassam](https://github.com/bassam))
+* Nic Cope <negz@upbound.io> ([negz](https://github.com/negz))
+* Jared Watts <jared@upbound.io> ([jbw976](https://github.com/jbw976))
+* Daniel Mangum <dan@upbound.io> ([hasheddan](https://github.com/hasheddan))
+* Muvaffak Onus <muvaf@upbound.io> ([muvaf](https://github.com/muvaf))
```
Comment (muvaf) [inline OWNERS.md:29]:
```suggestion
* Muvaffak Onus <monus@upbound.io> ([muvaf](https://github.com/muvaf))
```

Comment (jbw976) [reply OWNERS.md]: good catch @muvaf! I incorporated this change :)

**Comment by jbw976** [comment]
FYI, I plan to get current maintainer approval and merge this PR on Tues May 26th

**Pull request merged**